src/test/javascript/spec/component/text-assessment-dashboard/text-assessment-dashboard.component.spec.ts

```diff
@@ -0,0 +1,253 @@
+
+const textExercise = {
+    id: 22,
+    course: { id: 91 },
+    type: ExerciseType.TEXT,
+    studentAssignedTeamIdComputed: true,
+    assessmentType: AssessmentType.SEMI_AUTOMATIC,
+    numberOfAssessmentsOfCorrectionRounds: [],
+    secondCorrectionEnabled: true,
+};
+const textExerciseOfExam = {
+    id: 23,
+    exerciseGroup: { id: 111, exam: { id: 112, course: { id: 91 } } },
+    type: ExerciseType.TEXT,
+    studentAssignedTeamIdComputed: true,
+    assessmentType: AssessmentType.SEMI_AUTOMATIC,
+    numberOfAssessmentsOfCorrectionRounds: [],
+    secondCorrectionEnabled: true,
+};
+const textSubmission = { id: 1, submitted: true, results: [{ id: 10, assessor: { id: 20, guidedTourSettings: [] } }], participation: { id: 1 } };
+
+describe('TextAssessmentDashboardComponent', () => {
+    let component: TextAssessmentDashboardComponent;
+    let fixture: ComponentFixture<TextAssessmentDashboardComponent>;
+    let exerciseService: ExerciseService;
+    let textSubmissionService: TextSubmissionService;
+    let textAssessmentService: TextAssessmentService;
+    let accountService: AccountService;
+    let sortService: SortService;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [
+                TextAssessmentDashboardComponent,
+                MockComponent(AssessmentFiltersComponent),
+                MockComponent(AssessmentWarningComponent),
+                MockComponent(ResultComponent),
+                MockDirective(HasAnyAuthorityDirective),
+                MockDirective(TranslateDirective),
+                MockDirective(SortDirective),
+                MockRouterLinkDirective,
+                MockQueryParamsDirective,
+                TranslatePipeMock,
+                MockPipe(ArtemisDatePipe),
+            ],
+            providers: [
+                JhiLanguageHelper,
+                { provide: AccountService, useClass: MockAccountService },
+                { provide: ExerciseService, useClass: MockExerciseService },
+                MockProvider(TextSubmissionService),
+                MockProvider(TextAssessmentService),
+                MockProvider(SortService),
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        snapshot: {
+                            paramMap: convertToParamMap({
+                                exerciseId: textExercise.id,
+                                examId: 2,
+                            }),
+                        },
+                    },
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(TextAssessmentDashboardComponent);
+                component = fixture.componentInstance;
+                exerciseService = fixture.debugElement.injector.get(ExerciseService);
```

Comment:
```suggestion
                exerciseService = TestBed.inject(ExerciseService);
```
We should avoid using injector.get if possible.